atom/renderer/api/atom_api_renderer_ipc.cc

```diff
@@ -46,23 +47,31 @@ void Send(mate::Arguments* args,
     args->ThrowError("Unable to send AtomViewHostMsg_Message");
 }
 
-base::string16 SendSync(mate::Arguments* args,
-                        const base::string16& channel,
-                        const base::ListValue& arguments) {
-  base::string16 json;
+v8::Local<v8::Value> SendSync(mate::Arguments* args,
+                              const base::string16& channel,
+                              const base::ListValue& arguments) {
+  base::ListValue result;
 
   RenderView* render_view = GetCurrentRenderView();
   if (render_view == nullptr)
-    return json;
+    return v8::Null(args->isolate());
 
   IPC::SyncMessage* message = new AtomViewHostMsg_Message_Sync(
-      render_view->GetRoutingID(), channel, arguments, &json);
+      render_view->GetRoutingID(), channel, arguments, &result);
   bool success = render_view->Send(message);
 
   if (!success)
     args->ThrowError("Unable to send AtomViewHostMsg_Message_Sync");
 
-  return json;
+  std::unique_ptr<atom::V8ValueConverter> v8_converter(
+      new atom::V8ValueConverter);
+  mate::Dictionary global(args->isolate(),
+                          args->isolate()->GetCurrentContext()->Global());
+  v8::Local<v8::Value> sandboxed;
+  if (global.GetHidden("sandboxed", &sandboxed) && sandboxed->IsTrue()) {
```

Comment:
Would it be better to use `v8::Context::[Set/Get]EmbedderData` for setting up this property per context rather accessing the global object ?